### ISSUE TITLE
docs: getting started guide

### ISSUE THREAD

**Issue by jeffwhelpley** [angular/universal#226]
This should be the very first thing anyone reads and it should be crazy simple. If it isn't, we have work to do to refine the API. This guide should include the following:

1. First a quickstart for building a simple hello, world universal app from scratch
1. Take existing client only Angular 2 app and make it universal
1. Best practices/guidelines in general when you are writing code to make it universal

**Comment by sumpton** [comment]
We would love any help. We also appreciate all the work that has gone into Universal and Angular.

We have a lot of Angular 1 in a large project nearing beta. We are wondering what we should be doing. We now see our existing code needs to be replaced and are, of course, considering React, in addition to Angular 2. We are a bit fearful of committing to Angular 2 at this time.

Universal is a big draw. We can ignore server-side rendering for another month or so, but it would be great to understand now how to best redevelop, how to hook in Universal, and how the stack should work with our Dust.js templates.

**Comment by jeffwhelpley** [comment]
So, regarding Dust.js or any other server side template system, you basically have two choices:

1. Only use Angular templates everywhere. With Universal there is no need for additional server side template engines, so you can just get rid of Dust.js altogether.
1. If you need/want to re-use your existing templates, you can still use another server template engine, but you basically will then be hooking up two server template engines. First, you use Dust.js or whatever as you do today, but then before you return the rendered HTML to the client, you would then use Universal to process the HTML further with Angular Universal.

I would need to learn more about your use case, but I would really strongly recommend going with option 1 if you can.

As for helping with docs, I am going to do some work on updating all the open issues tonight so I will let you know. In the meantime, if you send me your email address (jeffwhelpley at gmail), I can add you to our slack channel.

**Comment by sumpton** [comment]
Thank you for the clear response. We reached the same conclusion and have made the jump to Angular exclusively. Yesterday, we started refactoring a lot of Dust and Angular 1, and appreciate your validation. 

It took us a lot future guessing to not switch to React now, as it seems a safer bet. But we are drawn to the richness 2, and also the smaller, simpler, faster to develop and easier to maintain code we think we will have with 2 -- so thank you and everyone involved for all your clever work.
